scripts/lib/CIME/XML/generic_xml.py

```diff
@@ -289,6 +302,12 @@ def write(self, outfile=None, force_write=False):
         """
         Write an xml file from data in self
         """
+        timestamp_cache = self._FILEMAP[self.filename][2]
+        if timestamp_cache != 0.0:
+            timestamp_file  = os.path.getmtime(self.filename)
+            expect(timestamp_file == timestamp_cache,
```

Comment:
I haven't thought about how, if at all, this error-check interacts with my suggestions about changing where mtime is called.